app/controllers/candidate_interface/application_form_controller.rb

```diff
@@ -73,6 +74,16 @@ def review_submitted
       @application_form = current_application
     end
 
+    def review_previous_application
+      if @application_form.nil?
+        redirect_to candidate_interface_application_form_path
+      else
+        @review_previous_application = true
+
+        render :review_submitted
+      end
+    end
```

Comment:
As a fix for my other comments, what would you make of something like this?

```suggestion
    def review_previous_application
      @application_form = current_candidate.application_forms.find(params[:id])
      @review_previous_application = true
      render :review_submitted
    rescue ActiveRecord::RecordNotFound
      render_404
    end
```

Reply:
Looks good to me 👍 The rescue should never be hit right other than manually messing with the URL, but it's still good to have in place!